### ISSUE TITLE
Clarity on Understanding 1.4.11 regarding hue

### ISSUE THREAD

**Issue by fstorr** [w3c/wcag#868]
Hi

I'm trying to understand the pass/fail examples in the [two-star-ratings example](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html).

The first failure example (the stars with the black borders) states it fails because it relies on yellow and white hues. But I can't get from white to yellow just by altering hue—I have to alter lightness. For example: hsl(60,100%,100%) is white, and hsl(60,100%,50%) is yellow. I haven't altered hue, I've altered lightness. The content headed "Relationship with Use of Color and Focus Visible" states "the principle is that contrast ratio (the difference in brightness) can be used to distinguish text or graphics". As I've just altered the lightness, shouldn't this pass?

I've also looked at [Using a contrast ratio of 3:1 with surrounding text and providing additional visual cues on focus for links or controls where color alone is used to identify them](https://www.w3.org/WAI/WCAG21/Techniques/general/G183) which mentions "a relative luminance (lightness)" (nb: the link on 'relative luminance' is broken).

Am I missing something, and if so, can someone clarify the point for me.

Thanks

**Comment by mraccess77** [comment]
My personal view is that the change in lightness is not > 3:1 ratio and thus the primary differentiation is the hue change seen in yellow and white even if there is a small change in lightness.    Perhaps that is what should be said.

**Comment by JAWS-test** [comment]
I would say that example violates 1.4.11 because white and yellow do not have sufficient contrast distance (less than 3:1).

I can't see a violation of 1.4.1 because the colors don't transport information (it doesn't matter which color is used, it's just that some stars are filled and others aren't, and if I can't distinguish this it's just 1.4.11). 

"The first fails the Use of color criteria due to relying on yellow and white hues" is false, IMO.

**Comment by fstorr** [comment]
I would say that the stars do transport information—it's something like a product rating, so the information would be that the product has a rating of 2 out of 5. If the review system allowed fractions, then a star could be partially filled with white and yellow.

**Comment by JAWS-test** [comment]
> I would say that the stars do transport information

Of course, the stars carry information. I also assume that it's about a rating (2 out of 5). But: It is not the colors yellow and white that transport the information, but the difference of the colors. It could also be blue and white or green and red. That's why it's not the colors that matter, but the noticeable difference between the 2 colors - and that's the contrast, as I wrote.

**Comment by JAWS-test** [comment]
Color coding according to 1.4.1 would be if, for example, green stars were used for a good rating and red stars for a bad rating. Then it would be a social convention that the color green stands for good and red for bad. And if there were no visual means other than the colours to transport the information, this would be a violation of 1.4.1, even if the contrasts of red and green were sufficient. If the contrasts of red and green were not sufficient, it would be a violation of 1.4.1 and 1.4.11.

**Comment by mraccess77** [comment]
there is precedent in using SC 1.4.1 for situations where it's not a specific color but any perceived hue/color difference used to communicate meaning without sufficient contrast (3:1)
G183: Using a contrast ratio of 3:1 with surrounding text and providing additional visual cues on focus for links or controls where color alone is used to identify them is mapped to SC 1.4.1

**Comment by JAWS-test** [comment]
@mraccess77 : I understood that the precedent only existed because WCAG 2.0 did not check the contrast of non-text content. G183 was then created to compensate for this deficiency. With the new WCAG 2.1 and according to logic this belongs to 1.4.11 and no longer to 1.4.1.

**Comment by JAWS-test** [comment]
Apart from the fact that the explanations in 1.4.1 are contradictory anyway.

On the one hand
> This success criterion addresses color perception specifically. Other forms of perception are covered in Guideline 1.3 including programmatic access to color and other visual presentation coding. 
On the other hand
> Benefits: People using Braille displays or other tactile interfaces can detect text cues by touch.

**Comment by patrickhlauke** [comment]
> I understood that the precedent only existed because WCAG 2.0 did not check the contrast of non-text content. G183 was then created to compensate for this deficiency. With the new WCAG 2.1 and according to logic this belongs to 1.4.11 and no longer to 1.4.1.

1.4.11 relates to *adjacent* colors ... colors that are right next to each other/are touching. it does *not* relate to difference in color of different states of a control/icon/etc. there doesn't need to be a 3:1 ratio between the yellow and white fill of the star rating, as the yellow and the white are not adjacent, they're two separate states.

hency, this does still only fall under 1.4.1 as it's about using color alone to differentiate between a "full" and an "empty" star.

**Comment by JAWS-test** [comment]
@patrickhlauke: Understood, thanks for the explanation.

**Comment by alastc** [comment]
Hi @fstorr, does the thread above answer your question?

**Comment by alastc** [comment]
@fstorr wrote:
> The first failure example (the stars with the black borders) states it fails because it relies on yellow and white hues. But I can't get from white to yellow just by altering hue—I have to alter lightness... As I've just altered the lightness, shouldn't this pass?

Proposed response:
<hr>
The caption for the example says, the first "fails the **Use of color** criteria due to relying on yellow and white hues." 

There is a small difference in lightness as well, but it is not a 3:1 level of difference, therefore contrast is not being used to differentiate the full/empty stars.
 
Contrast can be a differentiator (as G183 does for text links), however, you would need to use a luminance different of 3:1 or more.

**Comment by alastc** [comment]
In the meeting yesterday we agreed to accept the reponse above, and update the text of the response in @fstorr's branch:
https://www.w3.org/2020/08/04-ag-minutes.html#item06

The text from the meeting was:
> Figure 7 Two examples which fail a success criteria, the first fails the Use of color criteria due to relying on yellow and white hues. The second example fails the Non-text contrast criteria due to the yellow (#fff000) against white background of 1.2:1.

But if @gundulaniemann and @mbgower want to agree an update on that, please go ahead.

**Comment by patrickhlauke** [comment]
as noted on the actual commit/PR itself, that proposed text is grammatically/logically incorrect though "due to the yellow ... against white background of 1.2:1" makes no sense without mentioning it's the *contrast ratio* of 1.2:1.

**Comment by alastc** [comment]
Not sure why this one is still open, the current text for the example above is: 
"Two examples which fail a success criterion, the first fails the Use of color criterion due to relying on yellow and white hues. The second example fails the Non-text contrast criterion due to the yellow (#fff000) to white contrast ratio of 1.2:1."
